ggshield/scan/scannable.py

```diff
@@ -87,6 +87,10 @@ def scan_dict(self) -> Dict[str, Any]:
             "filemode": self.filemode,
         }
 
+    def has_extensions(self, extensions: Iterable[str]) -> bool:
+        """Returns True iff the file has one of the given extensions."""
+        return any(re.search(rf"\.{ext}$", self.filename) for ext in extensions)
```

Comment:
Instead of using a regex, you can make `extensions` a `Set[str]` and then the function can become something like this:
```python
_, extension = os.path.splitext(self.filename)
return extension in extensions
```

Reply:
Yes, I was doing something like that at first but saw what was already done in one of the internal projects. No preference on my side, I will do this since it introduces less dependencies